### DISCUSSION TITLE
Filtering out forks from search results

### DISCUSSION THREAD

**Discussion by rmax** [community/community#39870] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Is it possible to filter out forks from search results?

Edit: I think something like `foo NOT is:fork` would be nice

**Comment by david-engelmann** [comment]
you can use the fork:false flag to exclude forked results in search

**Reply by look** [reply]
This hasn't yet been implemented for the new code search. It's on our list!

**Comment by look** [comment]
@rmax thanks for the suggestion. We haven't implemented this yet, but I'm eager to add it. Please stay tuned.

**Comment by look** [comment, marked as answer]
Thanks for your patience. There is now an option to exclude forks (or limit your search to forks). It uses the syntax you suggested @rmax! 

To exclude forks from your results:

```
NOT is:fork
```

To search only fork repositories:

```
is:fork
```

For more information, please [see the documentation](https://docs.github.com/en/search-github/github-code-search/understanding-github-code-search-syntax#is-qualifier).

**Reply by clayne** [reply]
Docs aren't updated at all for this - including even the advanced search interface itself. It took me having to search for discussions on here to even find this.

Also, when using the REST API, one has to use `NOT fork:only` to get the same result. Things are terribly ambiguous right now between what the old/new interfaces support and additionally the dropping of the ability to sort by recently indexed+descending is forcing everyone looking only for new results to either a: query the entire result set then filter afterwards, b: get screwed completely if there's >1000 results.

**Reply by look** [reply]
> Docs aren't updated at all for this - including even the advanced search interface itself. It took me having to search for discussions on here to even find this.

Sorry about that. The support for including/excluding forks should be suggested by the autocomplete and is mentioned in our [documentation for the new search](https://docs.github.com/en/search-github/github-code-search/understanding-github-code-search-syntax#is-qualifier). The Advanced Search needs to be updated to reflect the new syntax when the viewer is opted into the the beta, we will fix that.


> Also, when using the REST API, one has to use `NOT fork:only` to get the same result. 

Thanks for bringing this up. This is a result of the new index including forks, whereas the old one usually did not. To make the results more comparable we are going to exclude forks from the API by default.

> Things are terribly ambiguous right now between what the old/new interfaces support and additionally the dropping of the ability to sort by recently indexed+descending is forcing everyone looking only for new results to either a: query the entire result set then filter afterwards, b: get screwed completely if there's >1000 results.

Thanks for the feedback. We will be releasing a new API that supports exhaustive result sets. Sorting by indexed at is not supported because it doesn't sort based on when a file was changed (which is what people need), but rather when a blob was indexed. This isn't a good proxy in the new system because we rebuild the search index constantly.

**Reply by clayne** [reply]
> Thanks for bringing this up. This is a result of the new index including forks, whereas the old one usually did not. To make the results more comparable we are going to exclude forks from the API by default.

Thanks for the reply. I too agree that excluding forks by default for the API side makes the most sense, particularly if the client side still has ability to control it such that they can include them if desired.

> Thanks for the feedback. We will be releasing a new API that supports exhaustive result sets. Sorting by indexed at is not supported because it doesn't sort based on when a file was changed (which is what people need), but rather when a blob was indexed. This isn't a good proxy in the new system because we rebuild the search index constantly.

Also good. I know that multiple people have brought this up in the larger beta discussion thread. Indeed, having the ability to sort by latest commit time of the repo would be the ideal - something we didn't even have in the older system given that indexed time was an alright proxy with the old approach.

Since we're here there's another issue affecting the REST API that I brought up in the larger discussion thread (but I doubt is going to be seen): REST API queries are now returning results w/ partial matches and without the ability to anchor tokens at all. This means if one were look for something like "sage" then anything including "message" will also be returned. This is very painful to deal with as the result sets are now an order of magnitude larger (depending on the searched term) and there is _no way to stop it_. Within the web UI one can use the new regex feature as a workaround to anchor via `/\Wsage\W/` but the API side simply does not support this at all. So presently various client side scripts I've written to look for new repos containing xyz terms are essentially dead in the water for certain search terms as the result set is too large to deal with. My comments related to this here: https://github.com/orgs/community/discussions/48301#discussioncomment-5689589
